components/string.rst

```diff
@@ -0,0 +1,451 @@
+    u('   Lorem Ipsum   ')->trimEnd();   // '   Lorem Ipsum'
+
+Methods to Search and Replace
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+::
+
+    // checks if the string starts/ends with the given string
+    u('https://symfony.com')->startsWith('https'); // true
+    u('report-1234.pdf')->endsWith('.pdf');        // true
+
+    // checks if the string contents are exactly the same as the given contents
+    u('foo')->equalsTo('foo'); // true
+
+    // checks if the string content match the given regular expression
+    u('avatar-73647.png')->match('/avatar-(\d+)\.png/');
+    // result = ['avatar-73647.png', '73647']
+
+    // finds the position of the first occurrence of the given string
+    // (the second argument is the position where the search starts and negative
+    // values have the same meaning as in PHP functions)
+    u('abcdeabcde')->indexOf('c');     // 2
+    u('abcdeabcde')->indexOf('c', 2);  // 2
+    u('abcdeabcde')->indexOf('c', -4); // 7
+    u('abcdeabcde')->indexOf('eab');   // 4
+    u('abcdeabcde')->indexOf('k');     // null
+
+    // finds the position of the last occurrence of the given string
+    // (the second argument is the position where the search starts and negative
+    // values have the same meaning as in PHP functions)
+    u('abcdeabcde')->indexOfLast('c');     // 7
+    u('abcdeabcde')->indexOfLast('c', 2);  // 7
+    u('abcdeabcde')->indexOfLast('c', -4); // 2
+    u('abcdeabcde')->indexOfLast('eab');   // 4
+    u('abcdeabcde')->indexOfLast('k');     // null
+
+    // replaces all occurrences of the given string
+    u('http://symfony.com')->replace('http://', 'https://'); // 'https://symfony.com'
+    // replaces all occurrences of the given regular expression
+    u('(+1) 206-555-0100')->replaceMatches('/[^A-Za-z0-9]++/', ''); // '12065550100'
+    // you can pass a callable as the second argument to perform advanced replacements
+    u('123')->replaceMatches('/\d/', function ($match) {
+        return '['.$match[0].']';
+    }); // result = '[1][2][3]'
+
+Methods to Join, Split and Truncate
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+::
+
+    // uses the string as the "glue" to merge all the given strings
+    u(', ')->join(['foo', 'bar']); // 'foo, bar'
+
+    // breaks the string into pieces using the given delimiter
+    u('template_name.html.twig')->split('.');    // ['template_name', 'html', 'twig']
+    // you can set the maximum number of pieces as the second argument
+    u('template_name.html.twig')->split('.', 2); // ['template_name', 'html.twig']
+
+    // returns a substring which starts at the first argument and has the length of the
+    // second optional argument (negative values have the same meaning as in PHP functions)
+    u('Symfony is great')->slice(0, 7);  // 'Symfony'
+    u('Symfony is great')->slice(0, -6); // 'Symfony is'
+    u('Symfony is great')->slice(11);    // 'great'
+    u('Symfony is great')->slice(-6);    // 'great'
```

Comment:
should be `' great'`

Reply:
Good catch! I changed the example to use `-5` so the result is `great`. Thanks.